homeassistant/components/netatmo/climate.py

```diff
@@ -3,71 +3,140 @@
+})
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_RELAY): cv.string,
-    vol.Optional(CONF_THERMOSTAT, default=[]):
-        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_HOMES): vol.All(cv.ensure_list, [HOME_CONFIG_SCHEMA])
 })
 
+STATE_NETATMO_SCHEDULE = 'schedule'
+STATE_NETATMO_HG = 'hg'
+STATE_NETATMO_MAX = 'max'
+STATE_NETATMO_AWAY = 'away'
+STATE_NETATMO_OFF = STATE_OFF
+STATE_NETATMO_MANUAL = STATE_MANUAL
+
+DICT_NETATMO_TO_HA = {
+    STATE_NETATMO_SCHEDULE: STATE_AUTO,
+    STATE_NETATMO_HG: STATE_COOL,
+    STATE_NETATMO_MAX: STATE_HEAT,
+    STATE_NETATMO_AWAY: STATE_ECO,
+    STATE_NETATMO_OFF: STATE_OFF,
+    STATE_NETATMO_MANUAL: STATE_MANUAL
+}
+
+DICT_HA_TO_NETATMO = {
+    STATE_AUTO: STATE_NETATMO_SCHEDULE,
+    STATE_COOL: STATE_NETATMO_HG,
+    STATE_HEAT: STATE_NETATMO_MAX,
+    STATE_ECO: STATE_NETATMO_AWAY,
+    STATE_OFF: STATE_NETATMO_OFF,
+    STATE_MANUAL: STATE_NETATMO_MANUAL
+}
+
 SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
                  SUPPORT_AWAY_MODE)
 
+NA_THERM = 'NATherm1'
+NA_VALVE = 'NRV'
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the NetAtmo Thermostat."""
     netatmo = hass.components.netatmo
-    device = config.get(CONF_RELAY)
 
     import pyatmo
+    homes_conf = config.get(CONF_HOMES)
     try:
-        data = ThermostatData(netatmo.NETATMO_AUTH, device)
-        for module_name in data.get_module_names():
-            if CONF_THERMOSTAT in config:
-                if config[CONF_THERMOSTAT] != [] and \
-                   module_name not in config[CONF_THERMOSTAT]:
-                    continue
-            add_entities([NetatmoThermostat(data, module_name)], True)
+        home_data = HomeData(netatmo.NETATMO_AUTH)
     except pyatmo.NoDevice:
-        return None
+        return
+
+    homes = []
+    rooms = {}
+    if homes_conf is not None:
+        for home_conf in homes_conf:
+            home = home_conf[CONF_NAME]
+            if home_conf[CONF_ROOMS] != []:
+                rooms[home] = home_conf[CONF_ROOMS]
+            homes.append(home)
+    else:
+        homes = home_data.get_home_names()
+
+    for home in homes:
+        _LOGGER.debug("Setting up %s ...", home)
+        try:
+            room_data = ThermostatData(netatmo.NETATMO_AUTH, home)
+        except pyatmo.NoDevice:
+            continue
+        for room_id in room_data.get_room_ids():
+            room_name = room_data.homedata.rooms[home][room_id]['name']
+            _LOGGER.debug("Setting up %s (%s) ...", room_name, room_id)
+            if home in rooms and room_name not in rooms[home]:
+                _LOGGER.debug("Excluding %s ...", room_name)
+                continue
+            _LOGGER.debug("Adding devices for room %s (%s) ...",
+                          room_name, room_id)
+            add_entities([NetatmoThermostat(room_data, room_id)], True)
```

Comment:
Add all entities in one call to `add_entities`.

Reply:
I agree, I will create another PR to include this.

Reply:
Modified in https://github.com/home-assistant/home-assistant/pull/21633.